code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -264,3 +264,278 @@
 
 #undef RIGHT_HAND
 #undef LEFT_HAND
+
+/obj/item/organ/internal/cyberimp/arm/katana
+	name = "dark shard"
+	desc = "An eerie metal shard surrounded by dark energies."
+	icon = 'icons/obj/lavaland/artefacts.dmi'
+	icon_state = "cursed_katana_organ"
+	status = 0
+	contents = newlist(/obj/item/cursed_katana)
+
+
+/obj/item/organ/internal/cyberimp/arm/katana/attack_self(mob/living/carbon/user, modifiers)
+	. = ..()
+	to_chat(user,"<span class='userdanger'>The mass goes up your arm and inside it!</span>")
+	playsound(user, 'sound/misc/demon_consume.ogg', 50, TRUE)
+	user.apply_status_effect(STATUS_EFFECT_KATANA_CURSE)
+	user.drop_item()
+	insert(user)
+
+/obj/item/organ/internal/cyberimp/arm/katana/emp_act() //Organic, no emp stuff
+	return
+
+/obj/item/organ/internal/cyberimp/arm/katana/Retract()
+	var/obj/item/cursed_katana/katana = holder
+	if(!katana || katana.shattered)
+		return FALSE
+	if(!katana.drew_blood)
+		to_chat(owner, "<span class='userdanger'>[katana] lashes out at you in hunger!</span>")
+		playsound(owner, 'sound/misc/demon_attack1.ogg', 50, TRUE)
+		if(parent_organ)
+			owner.apply_damage(25, BRUTE, parent_organ, TRUE)
+	katana.drew_blood = FALSE
+	katana.clean_blood()
+	return ..()
+
+/obj/item/organ/internal/cyberimp/arm/katana/Extend()
+	for(var/obj/item/cursed_katana/katana in contents)
+		if(katana.shattered)
+			to_chat(owner, "<span class='warning'> Your cursed katana has not reformed yet!</span>")
+			return FALSE
+	return ..()
+
+/obj/item/organ/internal/cyberimp/arm/katana/proc/user_death(mob/user)
+	Retract(TRUE)
+	user.visible_message("<span class='warning'>[user] begins to turn to dust, his soul being contained within [src]!</span>",
+		"<span class='userdanger'>You feel your body begin to turn to dust, your soul being drawn into [src]!</span>")
+	forceMove(get_turf(owner))
+	remove(user)
+	addtimer(CALLBACK(user, /mob/.proc/dust), 1 SECONDS)
```

Comment:
if you do this, whats the need for the status effect?

Reply:
See signal issues

Reply:
nevermind I am dumb, duplicate

Reply:
I am twice as dumb and should not read code I wrote 2 weeks ago after waking up. Called by the debuff